Source/WebGPU/WebGPU/Device.mm

```diff
@@ -47,6 +47,15 @@
 #import <wtf/StdLibExtras.h>
 #import <wtf/WeakPtr.h>
 
+#if USE(INTERNAL_APPLE_SDK)
+#import <Metal/MTLResource_Private.h>
+#else
+@protocol MTLResourceSPI <MTLResource>
```

Comment:
Usually we put these in their own MetalSPI.h file

Reply:
Should we introduce a new MetalSPI.h for WebGPU.framework? I don't think one exists yet

Reply:
I didn't add one since I don't think a single SPI usage warrants its own file

Reply:
Yes, we should make a MetalSPI. It's important we keep accounting of SPI clear.

Reply:
SPI header added :)